hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieMergeHandle.java

```diff
@@ -294,6 +294,8 @@ protected boolean writeRecord(HoodieRecord<T> hoodieRecord, Option<IndexedRecord
         // Convert GenericRecord to GenericRecord with hoodie commit metadata in schema
         IndexedRecord recordWithMetadataInSchema = rewriteRecord((GenericRecord) indexedRecord.get());
         if (preserveMetadata) {
+          // do not preserve FILENAME_METADATA_FIELD
```

Comment:
is it always the case where we never preserve FILENAME ? since when compact we always create new files, right?  then it's some extra rule we have to remember for this config, which sounds like we preserve all meta cols.

Reply:
i don't have a better idea. just felt the confusion here.

Reply:
yes. so by preserve commit metadata, implicitly we mean to preserve every commit metadata except filename. Every other meta field should be carry forwarded from old record, but filename has to represent the file where the record actually exists.